connect/api/src/test/java/org/apache/kafka/connect/data/SchemaBuilderTest.java

```diff
@@ -36,6 +38,255 @@ public class SchemaBuilderTest {
     private static final String DOC = "doc";
     private static final Map<String, String> NO_PARAMS = null;
 
+    @Test
+    public void testDefaultValueStructSchema() {
+        SchemaBuilder builder = SchemaBuilder.struct()
+                .field("f1", Schema.BOOLEAN_SCHEMA);
+
+        Struct defaultValue = new Struct(builder.build()); // the Struct receives a schema, not a builder
+        defaultValue.put("f1", true);
+
+        builder.defaultValue(defaultValue)
+                .build();
```

Comment:
here we end up in a situation where defaultValue.schema().equals(builder.build()) is false - is this acceptable?
SchemaBuilder.defaultValue validates first, then sets the defaultValue field, that's why the builder.defaultValue call on line 49 passes, but I'm not sure if the result is desirable